Flow.Launcher.Core/Plugin/JsonRPCV2Models/JsonRPCPublicAPI.cs

```diff
@@ -0,0 +1,326 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Flow.Launcher.Plugin;
+using Flow.Launcher.Plugin.SharedModels;
+
+namespace Flow.Launcher.Core.Plugin.JsonRPCV2Models
+{
+    public class JsonRPCPublicAPI
+    {
+        private IPublicAPI _api;
+
+        public JsonRPCPublicAPI(IPublicAPI api)
+        {
+            _api = api;
+        }
+
+        /// <summary>
+        /// Change Flow.Launcher query
+        /// </summary>
+        /// <param name="query">query text</param>
+        /// <param name="requery">
+        /// Force requery. By default, Flow Launcher will not fire query if your query is same with existing one. 
+        /// Set this to <see langword="true"/> to force Flow Launcher requerying
+        /// </param>
+        public void ChangeQuery(string query, bool requery = false)
+        {
+            _api.ChangeQuery(query, requery);
+        }
+
+        /// <summary>
+        /// Restart Flow Launcher
+        /// </summary>
+        public void RestartApp()
+        {
+            _api.RestartApp();
+        }
+
+        /// <summary>
+        /// Run a shell command
+        /// </summary>
+        /// <param name="cmd">The command or program to run</param>
+        /// <param name="filename">the shell type to run, e.g. powershell.exe</param>
+        /// <exception cref="FileNotFoundException">Thrown when unable to find the file specified in the command </exception>
+        /// <exception cref="Win32Exception">Thrown when error occurs during the execution of the command </exception>
```

Comment:
@taooceros why are we duplicating a lot of the same method summaries here from IPublicAPI? Can we just keep them in the interface only.

Reply:
The problem is we cannot use `IPublicAPI` directly because there's an event handler there.

Reply:
Oh you are saying the summary I guess I just copy them from there... I can remove them